tokio/src/runtime/local_runtime/runtime.rs

```diff
@@ -0,0 +1,376 @@
+#![allow(irrefutable_let_patterns)]
+
+use crate::runtime::blocking::BlockingPool;
+use crate::runtime::scheduler::CurrentThread;
+use crate::runtime::{context, Builder, EnterGuard, Handle, BOX_FUTURE_THRESHOLD};
+use crate::task::JoinHandle;
+
+use std::future::Future;
+use std::marker::PhantomData;
+use std::time::Duration;
+
+/// A local Tokio runtime.
+///
+/// This runtime is capable of driving tasks which are not `Send + Sync` without the use of a
+/// `LocalSet`, and thus supports `spawn_local` without the need for a `LocalSet` context.
+///
+/// This runtime cannot be moved between threads or driven from different threads.
+///
+/// This runtime is incompatible with `LocalSet`. You should not attempt to drive a `LocalSet` within a
+/// `LocalRuntime`.
```

Comment:
LocalRuntime will intercept `spawn_local` calls within the localset context, and get the events.

I didn't want to specify the exact behavior here, in case we make changes such that this panics or similar.